File: resize_images.rb
require 'fileutils'
require 'mini_magick'

# Define directories
base_dir = File.dirname(__FILE__)
big_dir = File.join(base_dir, 'assets', 'images', 'releases', 'big')
small_dir = File.join(base_dir, 'assets', 'images', 'releases', 'small')

# Ensure small directory exists
FileUtils.mkdir_p(small_dir)

# Clean small directory
Dir.glob(File.join(small_dir, '*')).each do |f|
  File.delete(f) if File.file?(f)
end

# Process each big image
Dir.glob(File.join(big_dir, '*')).each do |img_path|
  next unless File.file?(img_path)
  
  begin
    # Get file name and create small name
    filename = File.basename(img_path)
    name = File.basename(filename, '.*')
    ext = File.extname(filename)
    
    # Replace _big with _small in filename
    if name.include?('_big')
      small_name = name.gsub('_big', '_small')
    else
      small_name = "#{name}_small"
    end
    
    small_path = File.join(small_dir, "#{small_name}#{ext}")
    
    # Resize image
    image = MiniMagick::Image.open(img_path)
    
    # Calculate new dimensions (500px width)
    width = 500
    height = (image.height.to_f * (width.to_f / image.width.to_f)).to_i
    
    # Resize and save
    image.resize "#{width}x#{height}"
    image.write small_path
    
    puts "Resized: #{filename} -> #{small_name}#{ext}"
  rescue => e
    puts "Error processing #{filename}: #{e.message}"
  end
end

puts "Image resizing completed."

File: resize_images.js
const fs = require('fs');
const path = require('path');
const sharp = require('sharp');

// Define directories
const baseDir = __dirname;
const bigDir = path.join(baseDir, 'assets', 'images', 'releases', 'big');
const smallDir = path.join(baseDir, 'assets', 'images', 'releases', 'small');

console.log(`Script running from: ${baseDir}`);
console.log(`Big directory path: ${bigDir}`);
console.log(`Small directory path: ${smallDir}`);

// Check directory access
try {
  // Check if we can access the big directory
  fs.accessSync(bigDir, fs.constants.R_OK);
  console.log('Big directory is accessible for reading');
} catch (err) {
  console.error(`Cannot access big directory: ${err.message}`);
}

// Ensure small directory exists
try {
  if (!fs.existsSync(smallDir)) {
    fs.mkdirSync(smallDir, { recursive: true });
    console.log('Created small directory');
  } else {
    console.log('Small directory already exists');
    try {
      fs.accessSync(smallDir, fs.constants.W_OK);
      console.log('Small directory is writable');
    } catch (err) {
      console.error(`Cannot write to small directory: ${err.message}`);
    }
  }
} catch (err) {
  console.error(`Error with small directory: ${err.message}`);
}

// Clean small directory
try {
  const smallFiles = fs.readdirSync(smallDir);
  for (const file of smallFiles) {
    const filePath = path.join(smallDir, file);
    if (fs.statSync(filePath).isFile()) {
      fs.unlinkSync(filePath);
    }
  }
} catch (err) {
  console.error(`Error cleaning small directory: ${err.message}`);
}

// Process each big image
(async () => {
  try {
    console.log(`Looking for images in: ${bigDir}`);
    const files = fs.readdirSync(bigDir);
    console.log(`Found ${files.length} files in big directory`);
    
    for (const filename of files) {
      const imgPath = path.join(bigDir, filename);
      console.log(`Processing: ${filename}`);
      
      if (!fs.statSync(imgPath).isFile()) {
        console.log(`Skipping non-file: ${filename}`);
        continue;
      }
      
      try {
        // Get file name and create small name
        const name = path.parse(filename).name;
        const ext = path.parse(filename).ext;
        
        // Replace _big with _small in filename
        let smallName;
        if (name.includes('_big')) {
          smallName = name.replace('_big', '_small');
        } else {
          smallName = `${name}_small`;
        }
        
        const smallPath = path.join(smallDir, `${smallName}${ext}`);
        
        // Get image metadata
        const metadata = await sharp(imgPath).metadata();
          // Calculate new dimensions (500px width)
        const width = 500;
        const height = Math.round(metadata.height * (width / metadata.width));
        console.log(`Resizing to ${width}x${height}`);
        
        // Resize and save
        await sharp(imgPath)
          .resize(width, height)
          .toFile(smallPath);
        
        console.log(`Resized: ${filename} -> ${smallName}${ext}`);
        
        // Verify the file was created
        if (fs.existsSync(smallPath)) {
          console.log(`Verified: ${smallName}${ext} was created successfully`);
        } else {
          console.log(`Warning: ${smallName}${ext} was not created`);
        }
      } catch (e) {
        console.error(`Error processing ${filename}: ${e.message}`);
      }
    }
    
    console.log("Image resizing completed.");
  } catch (err) {
    console.error(`Error reading directory: ${err.message}`);
  }
})();
